cli/azd/pkg/commands/pipeline/pipeline.go

```diff
@@ -117,14 +117,27 @@ func folderExists(folderPath string) bool {
 	return false
 }
 
+func ymlExists(ymlPath string) bool {
+	info, err := os.Stat(ymlPath)
+	if os.IsNotExist(err) {
```

Comment:
If err is non nil, but not os.ErrNotExists (say there was some other IO error trying to stat the file) then `info` here will be the zero value.  This will mean that `info.Mode().IsRegular()` will probably return `true`, but that may not be what we expect.

`folderExists` treats any error from Stat as an indication the directory does not exist.  It feels like we should do something similar here and only return true from this method when err is non nil and `info.Mode().IsRegular()` returns true.

In practice, I suspect that 99.9% of the time this will be fine, since other IO errors will not be common.